File: knowledge/recursion/02_divide_and_Conquer/tower_of_hanoi/hanoi.py
def hanoi(n, A, B, C):
    if n == 1:
        print(A, '--->', C)
    else:
        hanoi(n-1, A, C, B)
        hanoi(1, A, B, C)
        hanoi(n-1, B, A, C)

if __name__ == "__main__":
    hanoi(3, 'A', 'B', 'C')

File: DSA/Recursion/Divide and Conquer/tower_of_hanoi/hanoi.cc
#include <stdio.h>

// src = source 來源
// dst = destination 目的地
void hanoi(int n, char buffer, char src, char dst) {
  printf("hanoi(%d, %c, %c, %c)\n", n, buffer, src, dst);
  if (n == 1)
    printf("move disk from %c to %c\n", src, dst);
  else  {
    hanoi(n - 1, dst, src, buffer);
    printf("move disk from %c to %c\n", src, dst);
    hanoi(n - 1, src, buffer, dst);
  }
}

int main() {
  int n;
  //scanf("%d", &n);
  hanoi(3, 'B', 'A', 'C');
  return 0;
}
